src/main/java/com/example/medicalclinic/model/Doctor.java

```diff
@@ -0,0 +1,56 @@
+package com.example.medicalclinic.model;
+
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import jakarta.persistence.*;
+import lombok.*;
+import java.util.*;
+
+@Getter
+@Setter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Entity
+@Table(name = "DOCTOR")
+public class Doctor {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    private String email;
+    private String password;
+    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
+    @JoinTable(
+            name = "doctor_facility",
+            joinColumns = @JoinColumn(name = "doctor_id"),
+            inverseJoinColumns = @JoinColumn(name = "facility_id")
+    )
+    @JsonBackReference
+    private Set<Facility> facilities = new HashSet<>();
+
+    public void updateFrom(Doctor other) {
+        if (other.getPassword() != null) {
```

Comment:
ofNullable(other.getPassword()).ifPresent(newPassword -> this.password = newPassword);
ofNullable(other.getEmail()).ifPresent(newEmail -> this.email= newEmail);